libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -772,6 +772,33 @@ void CompilerUtils::convertType(
 	Type::Category stackTypeCategory = _typeOnStack.category();
 	Type::Category targetTypeCategory = _targetType.category();
 
+	if (stackTypeCategory == Type::Category::UserDefinedValueType)
+	{
+		solAssert(_typeOnStack.isExplicitlyConvertibleTo(_targetType), "");
+		solAssert(_targetType.isValueType(), "");
+		return
+			convertType(
+				_targetType,
+				dynamic_cast<UserDefinedValueTypeType const&>(_typeOnStack).underlyingType(),
```

Comment:
shouldn't this be
```suggestion
				dynamic_cast<UserDefinedValueTypeType const&>(_typeOnStack).underlyingType(),
				_targetType,
```
?